Source/Particles/PhotonParticleContainer.cpp

```diff
@@ -29,13 +29,22 @@ PhotonParticleContainer::PhotonParticleContainer (AmrCore* amr_core, int ispecie
     : PhysicalParticleContainer(amr_core, ispecies, name)
 {
 
+    m_my_type = PhysicalParticleType::photon;
```

Comment:
Could use `WarpXParticleContainer::physical_species` directly? It should be meant for that.

Reply:
ah, yes! sure...

Reply:
I've also replaced all the `AmISomething` functions with a unique: 
```
    template<PhysicalSpecies PhysSpec>
    bool AmIA () const noexcept {return (physical_species == PhysSpec);}
```
in `WarpXParticleContainer.H`

Reply:
That look nice.